Marshaling/unmarshaling of zero-sized table/record elements in bearer token

`TestFilterEncoding` works fine as it is now, but if you're to modify it this way:
```
--- a/pkg/token/bearer_test.go
+++ b/pkg/token/bearer_test.go
@@ -34,6 +34,7 @@ func TestBearerToken_Issuer(t *testing.T) {
 func TestFilterEncoding(t *testing.T) {
        f := token.NewBearerToken()
        f.SetLifetime(1, 2, 3)
+       f.SetEACLTable(eacl.NewTable())
 
        t.Run("binary", func(t *testing.T) {
                data, err := f.Marshal()
```
It'll fail with
```
--- FAIL: TestFilterEncoding (0.00s)
    --- FAIL: TestFilterEncoding/binary (0.00s)
        bearer_test.go:46: 
                Error Trace:    bearer_test.go:46
                Error:          Not equal: 
                                expected: &token.BearerToken{token:acl.BearerToken{body:(*acl.BearerTokenBody)(0xc000136018), sig:(*refs.Signature)(nil)}}
                                actual  : &token.BearerToken{token:acl.BearerToken{body:(*acl.BearerTokenBody)(0xc000136708), sig:(*refs.Signature)(nil)}}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -9,4 +9,3 @@
                                     cid: (*refs.ContainerID)(<nil>),
                                -    records: ([]*acl.Record) {
                                -    }
                                +    records: ([]*acl.Record) <nil>
                                    }),
                Test:           TestFilterEncoding/binary
    --- FAIL: TestFilterEncoding/json (0.00s)
        bearer_test.go:56: 
                Error Trace:    bearer_test.go:56
                Error:          Not equal: 
                                expected: &token.BearerToken{token:acl.BearerToken{body:(*acl.BearerTokenBody)(0xc000136018), sig:(*refs.Signature)(nil)}}
                                actual  : &token.BearerToken{token:acl.BearerToken{body:(*acl.BearerTokenBody)(0xc000136ae0), sig:(*refs.Signature)(nil)}}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -9,4 +9,3 @@
                                     cid: (*refs.ContainerID)(<nil>),
                                -    records: ([]*acl.Record) {
                                -    }
                                +    records: ([]*acl.Record) <nil>
                                    }),
                Test:           TestFilterEncoding/json
FAIL
```
You can add some records, but then it'll fail this way:
```
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -14,6 +14,4 @@
                                        action: (acl.Action) 1,
                                -       filters: ([]*acl.HeaderFilter) {
                                -       },
                                -       targets: ([]*acl.Target) {
                                -       }
                                +       filters: ([]*acl.HeaderFilter) <nil>,
                                +       targets: ([]*acl.Target) <nil>
                                       })
```
This doesn't happen with plain Table though, only with the one wrapped into bearer token and actually it seems that `SetEACLTable` is the culprit here, this test modification:
```
--- a/pkg/token/bearer_test.go
+++ b/pkg/token/bearer_test.go
@@ -34,6 +34,11 @@ func TestBearerToken_Issuer(t *testing.T) {
 func TestFilterEncoding(t *testing.T) {
        f := token.NewBearerToken()
        f.SetLifetime(1, 2, 3)
+       tab := eacl.NewTable()
+       require.Nil(t, tab.Records())
+       tab.AddRecord(eacl.CreateRecord(eacl.ActionDeny, eacl.OperationHead))
+       f.SetEACLTable(tab)
+       require.Nil(t, tab.Records())
 
        t.Run("binary", func(t *testing.T) {
                data, err := f.Marshal()
```
fails on the second `require.Nil`, so even though we've just asked bearer to use this table it actually has modified it. I'm not sure how to fix this properly, but this is what makes s3-gw tests fail with newer neofs-api-go at the moment (the bug didn't exist in v1.23.0, but now we have it in v1.26.1).

--- a/pkg/acl/eacl/filter.go
+++ b/pkg/acl/eacl/filter.go
@@ -66,7 +66,13 @@ func (f Filter) From() FilterHeaderType {
 }
 
 // ToV2 converts Filter to v2 acl.EACLRecord.Filter message.
+//
+// Nil Filter converts to nil.
 func (f *Filter) ToV2() *v2acl.HeaderFilter {
+	if f == nil {
+		return nil
+	}
+
 	filter := new(v2acl.HeaderFilter)
 	filter.SetValue(f.value.String())
 	filter.SetKey(f.key.String())

--- a/pkg/acl/eacl/filter.go
+++ b/pkg/acl/eacl/filter.go
@@ -108,6 +108,12 @@ func (k filterKey) String() string {
 }
 
 // NewFilter creates, initializes and returns blank Filter instance.
+//
+// Defaults:
+//  - header type: HeaderTypeUnknown;
+//  - matcher: MatchUnknown;
+//  - key: "";
+//  - value: "".
 func NewFilter() *Filter {
 	return NewFilterFromV2(new(v2acl.HeaderFilter))
 }

--- a/pkg/acl/eacl/filter_test.go
+++ b/pkg/acl/eacl/filter_test.go
@@ -59,3 +59,11 @@ func TestFilterEncoding(t *testing.T) {
 		require.Equal(t, f, d2)
 	})
 }
+
+func TestFilter_ToV2(t *testing.T) {
+	t.Run("nil", func(t *testing.T) {
+		var x *Filter
+
+		require.Nil(t, x.ToV2())
+	})
+}

--- a/pkg/acl/eacl/filter_test.go
+++ b/pkg/acl/eacl/filter_test.go
@@ -3,6 +3,7 @@ package eacl
 import (
 	"testing"
 
+	"github.com/nspcc-dev/neofs-api-go/v2/acl"
 	v2acl "github.com/nspcc-dev/neofs-api-go/v2/acl"
 	"github.com/stretchr/testify/require"
 )
@@ -65,5 +66,23 @@ func TestFilter_ToV2(t *testing.T) {
 		var x *Filter
 
 		require.Nil(t, x.ToV2())
+	})
+
+	t.Run("default values", func(t *testing.T) {
+		filter := NewFilter()
+
+		// check initial values
+		require.Empty(t, filter.Key())
+		require.Empty(t, filter.Value())
+		require.Equal(t, HeaderTypeUnknown, filter.From())
+		require.Equal(t, MatchUnknown, filter.Matcher())
+
+		// convert to v2 message
+		filterV2 := filter.ToV2()
+
+		require.Empty(t, filterV2.GetKey())
+		require.Empty(t, filterV2.GetValue())
+		require.Equal(t, acl.HeaderTypeUnknown, filterV2.GetHeaderType())
+		require.Equal(t, acl.MatchTypeUnknown, filterV2.GetMatchType())
 	})
 }

--- a/pkg/acl/eacl/record.go
+++ b/pkg/acl/eacl/record.go
@@ -135,22 +135,28 @@ func (r *Record) ToV2() *v2acl.Record {
 		return nil
 	}
 
-	targets := make([]*v2acl.Target, 0, len(r.targets))
-	for _, target := range r.targets {
-		targets = append(targets, target.ToV2())
-	}
-
-	filters := make([]*v2acl.HeaderFilter, 0, len(r.filters))
-	for _, filter := range r.filters {
-		filters = append(filters, filter.ToV2())
-	}
-
 	v2 := new(v2acl.Record)
+
+	if r.targets != nil {
+		targets := make([]*v2acl.Target, 0, len(r.targets))
+		for _, target := range r.targets {
+			targets = append(targets, target.ToV2())
+		}
+
+		v2.SetTargets(targets)
+	}
+
+	if r.filters != nil {
+		filters := make([]*v2acl.HeaderFilter, 0, len(r.filters))
+		for _, filter := range r.filters {
+			filters = append(filters, filter.ToV2())
+		}
+
+		v2.SetFilters(filters)
+	}
 
 	v2.SetAction(r.action.ToV2())
 	v2.SetOperation(r.operation.ToV2())
-	v2.SetTargets(targets)
-	v2.SetFilters(filters)
 
 	return v2
 }

--- a/pkg/acl/eacl/record.go
+++ b/pkg/acl/eacl/record.go
@@ -156,6 +156,12 @@ func (r *Record) ToV2() *v2acl.Record {
 }
 
 // NewRecord creates and returns blank Record instance.
+//
+// Defaults:
+//  - action: ActionUnknown;
+//  - operation: OperationUnknown;
+//  - targets: nil,
+//  - filters: nil.
 func NewRecord() *Record {
 	return new(Record)
 }

--- a/pkg/acl/eacl/record.go
+++ b/pkg/acl/eacl/record.go
@@ -128,7 +128,13 @@ func (r *Record) AddObjectOwnerIDFilter(m Match, id *owner.ID) {
 // TODO: add remaining filters after neofs-api#72
 
 // ToV2 converts Record to v2 acl.EACLRecord message.
+//
+// Nil Record converts to nil.
 func (r *Record) ToV2() *v2acl.Record {
+	if r == nil {
+		return nil
+	}
+
 	targets := make([]*v2acl.Target, 0, len(r.targets))
 	for _, target := range r.targets {
 		targets = append(targets, target.ToV2())

--- a/pkg/acl/eacl/record_test.go
+++ b/pkg/acl/eacl/record_test.go
@@ -119,3 +119,11 @@ func TestRecordEncoding(t *testing.T) {
 		require.Equal(t, r, r2)
 	})
 }
+
+func TestRecord_ToV2(t *testing.T) {
+	t.Run("nil", func(t *testing.T) {
+		var x *Record
+
+		require.Nil(t, x.ToV2())
+	})
+}

--- a/pkg/acl/eacl/record_test.go
+++ b/pkg/acl/eacl/record_test.go
@@ -126,4 +126,22 @@ func TestRecord_ToV2(t *testing.T) {
 
 		require.Nil(t, x.ToV2())
 	})
+
+	t.Run("default values", func(t *testing.T) {
+		record := NewRecord()
+
+		// check initial values
+		require.Equal(t, OperationUnknown, record.Operation())
+		require.Equal(t, ActionUnknown, record.Action())
+		require.Nil(t, record.Targets())
+		require.Nil(t, record.Filters())
+
+		// convert to v2 message
+		recordV2 := record.ToV2()
+
+		require.Equal(t, v2acl.OperationUnknown, recordV2.GetOperation())
+		require.Equal(t, v2acl.ActionUnknown, recordV2.GetAction())
+		require.Nil(t, recordV2.GetTargets())
+		require.Nil(t, recordV2.GetFilters())
+	})
 }

--- a/pkg/acl/eacl/table.go
+++ b/pkg/acl/eacl/table.go
@@ -88,13 +88,16 @@ func (t *Table) ToV2() *v2acl.Table {
 		v2.SetContainerID(t.cid.ToV2())
 	}
 
-	records := make([]*v2acl.Record, 0, len(t.records))
-	for _, record := range t.records {
-		records = append(records, record.ToV2())
+	if t.records != nil {
+		records := make([]*v2acl.Record, 0, len(t.records))
+		for _, record := range t.records {
+			records = append(records, record.ToV2())
+		}
+
+		v2.SetRecords(records)
 	}
 
 	v2.SetVersion(t.version.ToV2())
-	v2.SetRecords(records)
 
 	return v2
 }

--- a/pkg/acl/eacl/table.go
+++ b/pkg/acl/eacl/table.go
@@ -100,6 +100,13 @@ func (t *Table) ToV2() *v2acl.Table {
 }
 
 // NewTable creates, initializes and returns blank Table instance.
+//
+// Defaults:
+//  - version: pkg.SDKVersion();
+//  - container ID: nil;
+//  - records: nil;
+//  - session token: nil;
+//  - signature: nil.
 func NewTable() *Table {
 	t := new(Table)
 	t.SetVersion(*pkg.SDKVersion())

--- a/pkg/acl/eacl/table.go
+++ b/pkg/acl/eacl/table.go
@@ -75,7 +75,13 @@ func (t *Table) SetSignature(sig *pkg.Signature) {
 }
 
 // ToV2 converts Table to v2 acl.EACLTable message.
+//
+// Nil Table converts to nil.
 func (t *Table) ToV2() *v2acl.Table {
+	if t == nil {
+		return nil
+	}
+
 	v2 := new(v2acl.Table)
 
 	if t.cid != nil {

--- a/pkg/acl/eacl/table_test.go
+++ b/pkg/acl/eacl/table_test.go
@@ -119,4 +119,22 @@ func TestTable_ToV2(t *testing.T) {
 
 		require.Nil(t, x.ToV2())
 	})
+
+	t.Run("default values", func(t *testing.T) {
+		table := eacl.NewTable()
+
+		// check initial values
+		require.Equal(t, *pkg.SDKVersion(), table.Version())
+		require.Nil(t, table.Records())
+		require.Nil(t, table.CID())
+		require.Nil(t, table.SessionToken())
+		require.Nil(t, table.Signature())
+
+		// convert to v2 message
+		tableV2 := table.ToV2()
+
+		require.Equal(t, pkg.SDKVersion().ToV2(), tableV2.GetVersion())
+		require.Nil(t, tableV2.GetRecords())
+		require.Nil(t, tableV2.GetContainerID())
+	})
 }

--- a/pkg/acl/eacl/table_test.go
+++ b/pkg/acl/eacl/table_test.go
@@ -112,3 +112,11 @@ func TestTable_Signature(t *testing.T) {
 
 	require.Equal(t, sig, table.Signature())
 }
+
+func TestTable_ToV2(t *testing.T) {
+	t.Run("nil", func(t *testing.T) {
+		var x *eacl.Table
+
+		require.Nil(t, x.ToV2())
+	})
+}

--- a/pkg/acl/eacl/target.go
+++ b/pkg/acl/eacl/target.go
@@ -108,7 +108,13 @@ func (t Target) Role() Role {
 }
 
 // ToV2 converts Target to v2 acl.EACLRecord.Target message.
+//
+// Nil Target converts to nil.
 func (t *Target) ToV2() *v2acl.Target {
+	if t == nil {
+		return nil
+	}
+
 	target := new(v2acl.Target)
 
 	target.SetRole(t.role.ToV2())

--- a/pkg/acl/eacl/target.go
+++ b/pkg/acl/eacl/target.go
@@ -124,6 +124,10 @@ func (t *Target) ToV2() *v2acl.Target {
 }
 
 // NewTarget creates, initializes and returns blank Target instance.
+//
+// Defaults:
+//  - role: RoleUnknown;
+//  - keys: nil.
 func NewTarget() *Target {
 	return NewTargetFromV2(new(v2acl.Target))
 }

--- a/pkg/acl/eacl/target_test.go
+++ b/pkg/acl/eacl/target_test.go
@@ -61,3 +61,11 @@ func TestTargetEncoding(t *testing.T) {
 		require.Equal(t, tar, tar2)
 	})
 }
+
+func TestTarget_ToV2(t *testing.T) {
+	t.Run("nil", func(t *testing.T) {
+		var x *Target
+
+		require.Nil(t, x.ToV2())
+	})
+}

--- a/pkg/acl/eacl/target_test.go
+++ b/pkg/acl/eacl/target_test.go
@@ -4,6 +4,7 @@ import (
 	"crypto/ecdsa"
 	"testing"
 
+	"github.com/nspcc-dev/neofs-api-go/v2/acl"
 	v2acl "github.com/nspcc-dev/neofs-api-go/v2/acl"
 	crypto "github.com/nspcc-dev/neofs-crypto"
 	"github.com/nspcc-dev/neofs-crypto/test"
@@ -67,5 +68,19 @@ func TestTarget_ToV2(t *testing.T) {
 		var x *Target
 
 		require.Nil(t, x.ToV2())
+	})
+
+	t.Run("default values", func(t *testing.T) {
+		target := NewTarget()
+
+		// check initial values
+		require.Equal(t, RoleUnknown, target.Role())
+		require.Nil(t, target.BinaryKeys())
+
+		// convert to v2 message
+		targetV2 := target.ToV2()
+
+		require.Equal(t, acl.RoleUnknown, targetV2.GetRole())
+		require.Nil(t, targetV2.GetKeys())
 	})
 }

